### PR TITLE
Fix error that occurs when removing and reattaching map from document

### LINKED ISSUE #813: Detaching then reattaching <mapml-viewer> fails
If you try to remove the viewer from the document, and then re-attach it to the DOM, you get an error that results from the element lifecycle callback trying to re-create the shadow root. 

image

### PR DESCRIPTION
**Pull request by prushforth**
Closes #813

### CODE REVIEW

**Comment by prushforth** [comment]
This is a bit tricky to test; couldn't figure out how to make it fail on main and pass on this branch, because it throws in the browser on main, which upsets subsequent locators, waiting seemingly forever for events that don't happen due to the earlier thrown error.

File: web-map-index.html [hunk 1]
```diff
@@ -4,7 +4,7 @@
     <meta charset="utf-8">
     <meta name="viewport" content="width=device-width,initial-scale=1">
     <title>index-web-map.html</title>
-    <script type="module" src="dist/web-map.js"></script>
```
Comment (AliyanH) [inline web-map-index.html:7]: This does not work without "dist/"

**Review by AliyanH — APPROVED** [review]
LGTM!

**Comment by prushforth** [comment]
> LGTM!

Thanks for your help! Merging...

**Pull request merged**